doc/tae.rst

```diff
@@ -0,0 +1,50 @@
+.. _branin: quickstart.html#branin
+.. _spear: quickstart.html#spear-example
+
+.. _tae:
+
+Target Algorithm Evaluator
+--------------------------
+
+*SMAC* evaluates the algorithm to be optimized by invoking it through a Target
+Algortihm Evaluator (TAE). There are different TAEs implemented in *SMAC* which
+provide slightly different interfaces.
+
+- `smac.tae.execute_ta_run_old <apidoc/smac.tae.execute_ta_run_old.html#module-smac.tae.execute_ta_run_old>`_:
+  Used when calling *SMAC* via the `commandline <basic_usage.html#commandline>`_, where it takes the algorithm call
+  from the algo-parameter of the `scenario <options.html#scenario>`_.
+  The parameters will be appended to the algorithm call, which in total looks
+  like:
+
+  .. code-block:: bash
+
+      <algo> <instance> <instance specific> <cutoff time> <runlength> <seed> <algorithm parameters>
+      python algo.py 123 0 10 25 12345 -param1 value1 -param2 value2 [...]
+
+  It expects the target algorithm to print a string during
+  execution with on of the following formats:
+
+  .. code-block:: bash
+
+      Result of this algorithm run: <STATUS>, <running time>, <runlength>, <quality>, <seed>, <instance-specifics>
+      Result for SMAC: <STATUS>, <running time>, <runlength>, <quality>, <seed>, <instance-specifics>
+      Result for ParamILS: <STATUS>, <running time>, <runlength>, <quality>, <seed>, <instance-specifics>
+
+  The examples Branin_ or SPEAR_ reflect the usage.
+  
+  * **STATUS** can be one of [SAT, UNSAT, SUCCESS, TIMEOUT, MEMOUT, CRASHED, ABORT]. SAT and UNSAT are mainly supported for backcompatibility and are treated as SUCCESS. The difference between CRASHED and ABORT is that ABORT is called when all future calls are assumed to crash and will abort the whole optimization, whereas CRASHED only indicates a single failed run.
+  * **running time** indicates the time that the execution took.
+  * **runlength** indicates the number of steps needed for execution.
+  * **quality** is the solution quality.
```

Comment:
Why are we actually calling this quality? SMAC is minimizing a loss/cost, so we should actually call this differently throughout the whole package.